Add dbt Fusion documentation (#1824)
astronomer/astronomer-cosmos: docs/configuration/cosmos-conf.rst

    - Default: ``True``
    - Environment Variable: ``AIRFLOW__COSMOS__ENABLE_CACHE_PROFILE``

.. _pre_dbt_fusion:

>     Since Cosmos 1.11, we introduced support to dbt Fusion. Some of the changes may not be compatible with legacy versions of dbt-core.

Reviewer: ```suggestion
    From Cosmos 1.11, we have introduced support for dbt Fusion. Some of the changes may not be compatible with legacy versions of dbt-core.
```

>     If you find any issues on how Cosmos interacts with older versions of dbt-core we introduced this configuration.

Reviewer: ```suggestion
    If you find any issues on how Cosmos interacts with older versions of dbt-core you can use this configuration.
```
